### PR TITLE
Regenerate Datastore Gapic to use new HTTP headers.

### PR DESCRIPTION
**Pull request by landrito**
Updates: https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1268

cc @swcloud @bjwatson

### CODE REVIEW

File: google-cloud-datastore/lib/google/cloud/datastore/version.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@
 module Google
   module Cloud
     module Datastore
-      VERSION = "0.24.0"
+      VERSION = "0.24.1"
```
Comment (blowmage) [inline google-cloud-datastore/lib/google/cloud/datastore/version.rb]: Please don't adjust version numbers. We do this as part of the release process.

Comment (landrito) [inline google-cloud-datastore/lib/google/cloud/datastore/version.rb]: Ah Gotcha. I'll do revert this the rest.

Comment (blowmage) [inline google-cloud-datastore/lib/google/cloud/datastore/version.rb]: Can you please remove this commit from the PR.

Comment (blowmage) [inline google-cloud-datastore/lib/google/cloud/datastore/version.rb]: Thanks!

**Pull request merged**